### PR TITLE
add experimenter role on create if not global role exists

### PR DESCRIPTION
**Pull request by mabiede**

### CODE REVIEW

File: pool/web/handler/admin_settings_actor_permissions.ml [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+      let read = Yojson.Safe.from_string %> Guard.ActorPermission.of_yojson in
+      CCResult.map_err
+        Pool_common.Message.authorization
+        (try read permission with
+         | _ -> Error "Undefined Yojson for actor permission.")
+    in
+    let events =
+      Cqrs_command.Guardian_command.DeleteActorPermission.handle ~tags
+    in
+    let handle = function
```
Comment (timohuber) [inline pool/web/handler/admin_settings_actor_permissions.ml:45]: Do you hide the error on purpose?

File: pool/web/view/page/page_admin_settings_actor_permissions.ml [hunk 2]
```diff
@@ -0,0 +1,136 @@
+open CCFun
```
Comment (timohuber) [inline pool/web/view/page/page_admin_settings_actor_permissions.ml:1]: Please add hint, as discussed.

File: pool/app/admin/repo.ml [hunk 3]
```diff
@@ -278,6 +278,50 @@ module Sql = struct
       Connection.exec request input)
     |> Utils.Database.exec_as_transaction (Pool_database.Label.value pool)
   ;;
+
+  let search_by_name_request ?conditions limit =
+    let default_contidion =
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/app/admin/repo.ml:283]: It's actually looking for name or email. I'll update the naming.

File: pool/web/handler/admin_experiments.ml [hunk 4]
```diff
@@ -186,12 +186,37 @@ let create req =
...
+        Lwt.both actor (actor |>> has_general_experimenter_permission)
+      with
+      | Ok actor, Ok false ->
+        ActorRole.create
+          ~target_uuid:(Uuid.target_of Experiment.Id.value id)
+          actor.Actor.uuid
+          `Experimenter
+        |> fun role ->
+        RolesGranted [ role ] |> Pool_event.guard |> CCList.return |> Lwt.return
+      | (Ok _ | Error _), (Ok _ | Error _) -> Lwt.return []
```
[the remark this answers is not in the document]
Comment (mabiede) [reply pool/web/handler/admin_experiments.ml:212]: simplified the check and added error handling for "NotFound Actor"

File: pool/web/handler/admin_settings_actor_permissions.ml (hunk 1 above)
Comment (mabiede) [reply pool/web/handler/admin_settings_actor_permissions.ml:45]: yes, here the error for the user is only "Permission denied"

**Review by timohuber — APPROVED** [review]

**Pull request merged**